modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/TypeNameParser.cs

```diff
@@ -30,19 +31,23 @@ internal class TypeNameParser
         private readonly int _start;
 
         /** */
-        private readonly string _typeName;
+        private char[] _typeNameArr;
+
+        /** */
+        private readonly bool _forceJavaNamingConventions;
 
         /** */
         private int _pos;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TypeNameParser" /> class.
         /// </summary>
-        private TypeNameParser(string typeName, ref int pos)
+        private TypeNameParser(string typeName, ref int pos, bool forceJavaNamingConventions)
         {
-            _typeName = typeName;
+            _typeNameArr = typeName.ToCharArray();
             _start = pos;
             _pos = _start;
+            _forceJavaNamingConventions = forceJavaNamingConventions;
```

Comment:
Separation of concerns: `TypeNameParser` should not care about `forceJavaNamingConventions`, it just parses the .NET type name. We should perform the `ToLower` call elsewhere.

Reply:
FIxed.